src/main/java/com/aws/greengrass/clientdevices/auth/certificate/CertificateStore.java

```diff
@@ -178,6 +178,28 @@ public  X509Certificate[] loadCaCertificateChain(URI privateKeyUri, URI certific
             throw new CertificateChainLoadingException("Unable to find aliases in the key manager with the given "
                     + "private key and certificate URIs");
         }
+
+        // TODO: We are doing this here to avoid changing the PKCS11 provider or the nucleus code base. It is not
+        //  pretty but
+        if (privateKeyUri.getScheme().equals("pkcs11")) {
```

Comment:
This is nasty it might be temporary. It is just to get reading certificates from HSM working without having to add this functionality to the PKCS11 provider, and SecurityService